### PR TITLE
Handle hashes as longs and minor refactors

### PR DESCRIPTION
**Pull request by SalusaSecondus**
Handle hashes as longs and minor refactors.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java [hunk 1]
```diff
@@ -36,17 +36,46 @@ abstract class EvpSignatureBase extends SignatureSpi {
...
         keyType_ = keyType;
         paddingType_ = paddingType;
         if (paddingType_ == RSA_PKCS1_PSS_PADDING) {
-            pssParams_ = PSSParameterSpec.DEFAULT;
+            internalSetParams(PSSParameterSpec.DEFAULT);
+            // digest_ is set by internalSetParameters
+        } else {
+            internalSetParams(null);
+            // Overwrite the 0 set by internalSetParameters
+            digest_ = digest;
```
Comment (WillChilds-Klein) [inline src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java:60]: don't we want this to be set to `0` if `paddingType_` isn't PSS? no other padding types take a digest.

Comment (SalusaSecondus) [reply src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java:60]: This is the digest used to hash the message directly. (e.g., SHA-256 for SHA256withRSA)

**Comment by SalusaSecondus** [comment]
Closing and reopening to poke the CI

File: src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java (hunk 1 above)
Comment (WillChilds-Klein) [reply src/com/amazon/corretto/crypto/provider/EvpSignatureBase.java:60]: ah, right.

**Pull request merged**